CMakeLists.txt

```diff
@@ -294,7 +294,21 @@ endif()
 # Required so velox code can be used in a dynamic library
 set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
 
-set(CMAKE_CXX_STANDARD 17)
+# For C++20 support we need GNU GCC11 (or later versions) or Clang/AppleClang 15
+# (or later versions) to get support for the used features.
```

Comment:
We could also check for specific the specific features we need (see [docs](https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html)) this would make this portable to other compilers. What are we using that's only in gcc 11? Afaik 10 should have full(?) C++20 support?

Reply:
gcc11 and clang15 have full C++20 support. gcc10 has only partial C++20 support. And one other reason was to set a minimum of compilers as a base from which we can build and check for features?

Reply:
I see there are a few features missing https://en.cppreference.com/w/cpp/compiler_support#cpp20

I am fine with setting a minimum version (should be documented in the readme if it's not) but this specific check will not trigger if someone uses the Intel compiler or something but I guess that would still throw an error once the compiler doesn't understand the C++20 flag so :shrug:

Reply:
I added a table in the readme. Please take a look.